roles/edpm_reboot/tasks/main.yaml

```diff
@@ -0,0 +1,69 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Make sure yum-utils is installed
+  ansible.builtin.package:
+    name: yum-utils
+
+- name: Check if reboot is required with needs-restarting
+  ansible.builtin.command: needs-restarting -r
+  register: needs_restarting_output
+  changed_when: false
+
+- name: Print return information from needs-restarting
+  ansible.builtin.debug:
+    var: needs_restarting_output.stdout
+
+- name: Create reboot file for needs-restarting
+  when: _no_reboot_message not in needs_restarting_output.stdout
+  block:
+    - name: Create a reboot_required directory if it does not exist
+      ansible.builtin.file:
+        path: /var/lib/openstack/reboot_required/
+        state: directory
+        mode: '0755'
+
+    - name: Create file for reboot required
+      ansible.builtin.file:
+        path: /var/lib/openstack/reboot_required/needs_restarting
+        state: touch
+        mode: '0600'
```

Comment:
The logic for using needs-restarting to determine if a reboot is required feels slightly off. The man page says:

``
       needs-restarting  looks  through running processes and tries to detect those that use files from packages that have been updated after the given process started.
       Such processes are reported by this tool.
``

That says this tool is used to find processes that need to be restarted due to rpm updates. That does not imply a reboot is also required. I don't think we should reboot just because needs-restarting reports some processes might need to be restarted (perhaps there is a different role for that). Further most rpm's that update package and update systemd units will take care of restarting those using post scripts.

Reply:
The man page description says about processes restart, but in here I used it together with -r options. If you look at description of an option '-r' - it is used for checking if a full reboot is required because of kernel or core libraries updates. 

An example output of needs-restarting is:
```

root@server1:~> needs-restarting  -r ; echo $?
Core libraries or services have been updated:
  openssl-libs -> 1:1.0.1e-60.el7_3.1
  systemd -> 219-30.el7_3.9

Reboot is required to ensure that your system benefits from these updates.
```
We don't have role to update packages on compute nodes yet and that is why need-restarting command landed here, but maybe it could be moved from this role and be executed as a last step of a future package update role.

Reply:
I missed that you are using -r, sorry about that! I suppose it's possible some of the earlier package installs that could have happened during the bootstrap role may have update some of these core libraries.